src/theme/themeManager.ts

```diff
@@ -0,0 +1,58 @@
+export type CSSModuleClassNames = {
+	[key: string]: boolean;
+}
+
+export type AppliedClasses<T> = {
+	[P in keyof T]?: CSSModuleClassNames;
+}
+
+type Theme = {
+	[key: string]: string;
+}
+
+type CacheKey = {
+	baseThemeClasses: {};
+	overrideClasses?: {};
+};
+
+let currentTheme: Theme = {};
+let cacheMap = new Map<CacheKey, AppliedClasses<any>>();
+
+function addClassNameToMap(classMap: CSSModuleClassNames, classList: Theme, className: string) {
+	if (classList && classList.hasOwnProperty(className)) {
+		// split out the classname because css-module composition combines class names with a space
+		const generatedClassNames: string[] = classList[className].split(' ');
+		generatedClassNames.forEach((generatedClassName) => {
+			classMap[generatedClassName] = true;
+		});
+	}
+}
+
+export function setTheme(theme: {}) {
```

Comment:
I've tried that before, but then other parts start to fail, might be easier to ditch `Theme` and put `{}` for everything. Theme doesn't really give us anything.